lib/compat.php

```diff
@@ -322,3 +326,138 @@ function gutenberg_warn_classic_about_blocks() {
 		</script>
 	<?php
 }
+
+/**
+ * Adds a warning to the Classic Editor when CloudFlare is blocking REST API requests.
+ *
+ * @since 3.4.0
+ */
+function gutenberg_warn_classic_about_cloudflare() {
+	?>
+		<style type="text/css">
+			#cloudflare-block-dialog .notification-dialog {
+				padding: 25px;
+			}
+
+			#cloudflare-block-dialog ul {
+				list-style: initial;
+				padding-left: 20px;
+			}
+
+			@media only screen and (max-height: 480px), screen and (max-width: 450px) {
+				#cloudflare-block-dialog .notification-dialog {
+					width: 100%;
+					height: 100%;
+					max-height: 100%;
+					position: fixed;
+					top: 0;
+					margin: 0;
+					left: 0;
+				}
+			}
+		</style>
+
+		<div id="cloudflare-block-dialog" class="notification-dialog-wrap">
+			<div class="notification-dialog-background"></div>
+			<div class="notification-dialog">
+				<div class="cloudflare-block-message">
+					<h2><?php _e( 'Cloudflare is blocking REST API requests.', 'gutenberg' ); ?></h2>
+					<p><?php _e( 'Your site uses Cloudflare, which provides a Web Application Firewall (WAF) to secure your site against attacks. Unfortunately, some of these WAF rules are incorrectly blocking legitimate access to your site, preventing Gutenberg from functioning correctly.', 'gutenberg' ); ?></p>
+					<p><?php _e( "We're working closely with Cloudflare to fix this issue, but in the mean time, you can work around it in one of two ways:", 'gutenberg' ); ?></p>
+					<ul>
+						<li><?php _e( 'If you have a Cloudflare Pro account, login to Cloudflare, visit the Firewall settings page, open the "Cloudflare Rule Set" details, open the "Cloudflare WordPress" ruleset, then set the rules "WP0025A" and "WP0025B" to "Disable".', 'gutenberg' ); ?></li>
+						<li>
+						<?php
+							printf(
+								/* translators: link to a comment in the Gutenberg repo */
+								__( 'For free Cloudflare accounts, you can <a href="%s">change the REST API URL</a>, to avoid triggering the WAF rules. Please be aware that this may cause issues with other plugins that use the REST API, and removes any other protection Cloudflare may be offering for the REST API.', 'gutenberg' ),
+								'https://github.com/WordPress/gutenberg/issues/2704#issuecomment-410582252'
+							);
+						?>
+						</li>
+					</ul>
+					<p>
+					<?php
+						printf(
+							/* translators: link to an issue in the Gutenberg repo */
```

Comment:
Again missing the `%s` in the comment + I wouldn't shorten repository here.

`/* translators: %s: link to an issue in the Gutenberg GitHub repository */`